lightning-payjoin/src/lib.rs

```diff
@@ -1 +1,316 @@
+pub mod error;
 pub mod scheduler;
+
+use bitcoin::{psbt::Psbt, FeeRate, ScriptBuf, Transaction};
+use error::Error;
+use lightning::ln::ChannelId;
+
+use http_body_util::{BodyExt, Full};
+use hyper::body::{Body, Incoming};
+use hyper::header::HeaderValue;
+use hyper::server::conn::http1;
+use hyper::{HeaderMap, Request};
+use hyper_util::rt::TokioIo;
+use payjoin::receive::ProvisionalProposal;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::net::TcpStream;
+use tokio::sync::Mutex;
+
+/// Payjoin `Receiver` trait defines the behavior of the receiver side in a payjoin transaction
+/// withing a lightning network context.
+///
+/// Utilising payjoin transactions, a lightning node setup as a receiver can fund a channel from an
+/// incoming payjoin transaction saving additional on-chain transaction.
+pub trait Receiver: Send + Sync + 'static + Clone {
```

Comment:
This trait is implemented by `ldk-node` in the `payjoin_handler.rs` file. the following struct `LightningPayjoin` will use those functions to either accept normal pj transaction or open a channel